apps/backend/src/services/neighborhoodServices.ts

```diff
@@ -219,6 +257,76 @@ const connectUserToNeighborhood = async (userId: number, neighborhoodId: number)
   });
 };
 
+/**
+ * removes user from neighborhood in the db
+ * also, marks their open requests and responses in this neighborhood INACTIVE
+ * throws error if userId or neighborhoodId invalid
+ * throws error if user is not a member of the neighborhood
+ * @param userId
+ * @param neighborhoodId
+ */
+const removeUserFromNeighborhood = async (
+  userId: number,
+  neighborhoodId: number,
+): Promise<void> => {
+  const neighborhood = await prismaClient.neighborhood
+    .findUniqueOrThrow({
+      where: { id: neighborhoodId },
+      include: { users: true, requests: true },
+    });
+
+  const userIsMemberOfNeighborhood = neighborhood.users.some(user => user.id === userId);
+  const userIsNeighborhoodAdmin = neighborhood.admin_id === userId;
+
+  if (!userIsMemberOfNeighborhood) {
+    const error = new Error('User is not a member of this neighborhood.');
+    error.name = 'Unauthorized';
+    throw error;
+  } else if (userIsNeighborhoodAdmin) {
+    // In this case, prompt the user that this action will delete the neighborhood
+    // If confirmed, delete neighborhood
```

Comment:
We will have to issue another check on the frontend to see that the logged user is admin, right?